include/gatekeeper_lls.h

```diff
@@ -19,6 +19,69 @@
 #ifndef _GATEKEEPER_LLS_H_
 #define _GATEKEEPER_LLS_H_
 
-int run_lls(void);
+#include <rte_timer.h>
+
+#include "gatekeeper_mailbox.h"
+
+struct lls_cache {
+	/* Length (in bytes) of keys for this cache. */
+	uint32_t               key_len;
+
+	/* Timeout value (in seconds) to mark entries as stale. */
+	uint32_t               front_timeout_sec;
+	uint32_t               back_timeout_sec;
+
+	/* Name string needed for cache hash and mempools. */
+	const char             *name;
+
+	/* Array of cache entries indexed using @hash. */
+	struct lls_cache_entry **data;
+
+	/* Hash instance that maps IP address keys to LLS cache entries. */
+	struct rte_hash        *hash;
+
+	/* Memory pool for cache entries. */
+	struct rte_mempool     *entry_pool;
+
+	/* Mailbox to hold requests for changing the cache. */
+	struct mailbox         mb;
```

Comment:
I originally chose to associate a different mailbox with each cache because the requests for each cache will be of different lengths because the keys are different lengths, so the mempool would need to give out requests of diffferent sizes.

But instead, I've now made just one mailbox for LLS by having the mempool allocate requests with the maximum key size that will be used.